src/printer.rs

```diff
@@ -315,17 +310,102 @@ impl<'a> Printer for InteractivePrinter<'a> {
         };
 
         let description = &input.description;
+        let metadata = &input.metadata;
+
+        // We use this iterator to have a deterministic order for
+        // header components. HashSet has arbitrary order, but Vec is ordered.
+        let header_components: Vec<StyleComponent> = [
+            (
+                StyleComponent::HeaderFilename,
+                self.config.style_components.header_filename(),
+            ),
+            (
+                StyleComponent::HeaderFilesize,
+                self.config.style_components.header_filesize(),
+            ),
+            (
+                StyleComponent::HeaderPermissions,
+                self.config.style_components.header_permissions(),
+            ),
+            (
+                StyleComponent::HeaderLastModified,
+                self.config.style_components.header_last_modified(),
+            ),
+        ]
+        .iter()
+        .filter(|(_, is_enabled)| *is_enabled)
+        .map(|(component, _)| *component)
+        .collect();
+
+        header_components.iter().try_for_each(|component| {
+            if self.config.style_components.grid() {
+                write!(
+                    handle,
+                    "{}{}",
+                    " ".repeat(self.panel_width),
+                    self.colors
+                        .grid
+                        .paint(if self.panel_width > 0 { "│ " } else { "" }),
+                )?;
+            };
 
-        writeln!(
-            handle,
-            "{}{}{}",
-            description
-                .kind()
-                .map(|kind| format!("{}: ", kind))
-                .unwrap_or_else(|| "".into()),
-            self.colors.filename.paint(description.title()),
-            mode
-        )?;
+            match component {
+                StyleComponent::HeaderFilename => writeln!(
```

Comment:
It would be nice it we could de-duplicate the code in the match arms a bit

Reply:
After having removed permissions and lastmodified, I don't see much to de-duplicate at this point. Let me know if you have ideas about it.